spec/system/courses/students_spec.rb

```diff
@@ -55,6 +69,16 @@
       expect(page).to have_text(startup.founders.first.name)
     end
 
+    # Check the last seen for the first student
+    within("div[aria-label='Info of team #{team_1.name}']") do
+      expect(page).to have_text('Last seen 3 days ago')
```

Comment:
This test is insufficient.

Given that we must only update the `last_seen_at` once every _interval_ minutes, this delay in updates should be tested thoroughly using the time travel helper `travel_to`.

Reply:
@arpancodes: I think I know why the spec fails. First thing, after travelling to `10.mins` from now you need to refresh the page by using something like `visit current_path` and also do a `Load More..` as the team you are testing it on is the last one. 

But still this wouldn't work as for the browser the time is still the current system time and UI will display last seen 3 mins ago. So, the only way would be to inject a script and to push the browser time as well using something like [Sinon.JS](https://sinonjs.org/). Would that be required @harigopal ?

Reply:
> First thing, after travelling to 10.mins from now you need to refresh the page by using something like visit current_path

@arpancodes Yes, as @mahesh-krishnakumar indicated, you can't simply `travel_to` a future time, and then check the contents of the _previously_ loaded page - that doesn't actually make sense. You need to make another request for there to be any change _anywhere_...

But, on another note, I don't think we need to check what the _UI_ to confirm the time-spacing behaviour we've implemented, either.

What I mean is that the update of `set_last_seen_at` is a purely server-side event - the UI has no hand in it. What we should be testing is how frequently that value is being updated. This is a completely separate user feature that shouldn't be included in this particular spec file (because it has nothing to do with the list of students shown to a coach).

I've added a small spec that tests the basic functionality: https://github.com/pupilfirst/pupilfirst/pull/1024/commits/c0f3f13e3fb7ba7eeabde8232e9149a5a66cc10b

Reply:
This spec that we're commenting on will need to be cleaned up! 🙂